bpf/cpu/cpu.bpf.c

```diff
@@ -909,12 +943,33 @@ int walk_user_stacktrace_impl(struct bpf_perf_event_data *ctx) {
       return 1;
     }
 
+    u64 previous_rip = 0;
+
     // HACK(javierhonduco): This is an architectural shortcut we can take. As we
     // only support x86_64 at the minute, we can assume that the return address
     // is *always* 8 bytes ahead of the previous stack pointer.
-    u64 previous_rip_addr = previous_rsp - 8; // the saved return address is 8 bytes ahead of the previous stack pointer
-    u64 previous_rip = 0;
-    int err = bpf_probe_read_user(&previous_rip, 8, (void *)(previous_rip_addr));
+    #if __TARGET_ARCH_x86
+      u64 previous_rip_addr = previous_rsp - 8;
+      int err = bpf_probe_read_user(&previous_rip, 8, (void *)(previous_rip_addr));
+      if (err < 0) {
+        bpf_printk("\n FAILED TO READ PREVIOUS RIP: error: %d", err);
+      }
+      LOG("\tprevious ip: %llx (@ %llx)", previous_rip, previous_rip_addr);
+    #endif
+
+    #if __TARGET_ARCH_arm64
+      // For the leaf frame, the previous pc/ip is always be stored in the link register itself
```

Comment:
nit: 
```suggestion
      // For the leaf frame, the saved pc/ip is always be stored in the link register itself
```

I think previous is a bit ambiguous as we really mean the "next" one or, in this context, "saved return address"